src/sql/src/pure.rs

```diff
@@ -478,24 +479,87 @@ async fn purify_format_single(
             }
         }
         Format::Csv {
-            header_row,
             delimiter,
-            ..
+            ref mut columns,
         } => {
-            if *header_row && col_names.is_empty() {
-                if let Some(file) = file {
-                    let file = tokio::io::BufReader::new(file);
-                    let csv_header = file.lines().next_line().await?;
-                    match csv_header {
-                        Some(csv_header) => {
-                            csv_header
-                                .split(*delimiter as char)
-                                .for_each(|v| col_names.push(Ident::from(v)));
+            let first_row = if let Some(file) = file {
+                let file = tokio::io::BufReader::new(file);
+                let csv_header = file.lines().next_line().await;
+                match csv_header {
+                    Ok(Some(csv_header)) => Some(
+                        csv_header
+                            .split(*delimiter as char)
+                            .map(|name| name.to_string())
+                            .collect::<Vec<_>>(),
```

Comment:
I know this was here before but this does not seem like a valid way to parse a CSV header row? I think you need a proper CSV parser to account for the situation where the header names are double quoted.

Reply:
Yeah I'm planning on fixing that in a follow up PR, I should have added a TODO here to make that clear.

Reply:
The rust-csv crate seems to have a simple enough API, maybe it makes sense to just add this in this PR? Unless this fix is in a hurry it makes more sense to me to put a CSV parser here

Reply:
yeah the goal was to make the code review easier, not to delay the work, I'll just add it here since now it's the opposite.